src/library/coverart.cpp

```diff
@@ -28,20 +31,69 @@ QString typeToString(CoverInfo::Type type) {
     return "INVALID TYPE VALUE";
 }
 
+QString coverInfoRelativeToString(const CoverInfoRelative& infoRelative) {
+    return typeToString(infoRelative.type) % QLatin1Literal(",") %
+           sourceToString(infoRelative.source) % QLatin1Literal(",") %
+           infoRelative.coverLocation % QLatin1Literal(",") %
+           QLatin1Literal("0x") % QString::number(infoRelative.hash, 16);
+}
+
+QString coverInfoToString(const CoverInfo& info) {
+    return coverInfoRelativeToString(info) % QLatin1Literal(",") %
+           info.trackLocation % QLatin1Literal(",");
+}
+} // anonymous namespace
+
+
+bool operator==(const CoverInfoRelative& a, const CoverInfoRelative& b) {
+    return a.source == b.source &&
+            a.type == b.type &&
+            a.coverLocation == b.coverLocation &&
+            a.hash == b.hash;
```

Comment:
It's more efficient to compare the simple types and members which likely will differ first, i.e. hash -> source -> type -> coverLocation